src/frontend-pwa/src/views/Settings/Settings.tsx

```diff
@@ -185,26 +188,93 @@ export default function Settings() {
     }
   };
 
+  /**
+   * @summary Pulls in new map tile data if user hits the refresh button.
+   *          Forces the page to unregister the service-worker and reload
+   *          the window. This forces an updated service-worker to initialize
+   *          and download, triggering assets to be downloaded anew.
+   *
+   *
+   * @author  Dallas Richmond, Tyler Maloney
+   */
+  const handleMapTiles = () => {
+    if ('serviceWorker' in navigator) {
+      navigator.serviceWorker.getRegistrations().then((registrations) => {
+        registrations.forEach((registration) => {
+          registration.unregister();
+        });
+      });
+      navigator.serviceWorker.register('/sw.js')
+        .then(() => {
+          setMapsCache(true);
+          window.location.reload();
+        })
+        .catch((error) => {
+          console.error('Error registering service worker:', error);
+        });
+    }
+  };
+
+  /**
+   * @summary Show the map tile confirmation modal to the user.
+   * @author  Tyler Maloney
+   */
+  const handleClearCacheConfirm = () => {
+    setIsClearCacheModalOpen(true);
+  };
+
+  /**
+   * @summary Clears all mapTile data from browser cache,
+   *          sets the mapsCached state to false,
+   *          and closes the confirmation modal.
+   * @author  Tyler Maloney, Dallas Richmond
+   */
+  const handleCacheModalConfirm = () => {
+    if ('serviceWorker' in navigator) {
+      caches.keys()
+        .then((cacheNames) => {
+          const precacheName = cacheNames.filter((cacheName) => cacheName.startsWith('workbox-precache-v2'));
```

Comment:
I think with this file in particular, we would all benefit with some inline + more in-depth jsdoc comments to serve as future reference material. Especially given how much this part became a very high wall to climb

Reply:
honestly the issue was with me being lead down a hole involving event messaging, and thinking that was necessary for the service worker to communicate. once dallas started working with cache directly, the functionality became clearer. expanded upon it anyway